frontend/service.go

```diff
@@ -62,14 +62,15 @@ func (s *lwdStreamer) GetLatestBlock(ctx context.Context, placeholder *walletrpc
 	// Lock to ensure we return consistent height and hash
 	s.mutex.Lock()
 	defer s.mutex.Unlock()
-	latestBlock := s.cache.GetLatestHeight()
-
-	if latestBlock == -1 {
-		return nil, errors.New("cache is empty, server is probably not yet ready")
+	blockChainInfo, err := common.GetBlockChainInfo()
+	if err != nil {
+		return nil, err
 	}
-	latestHash := s.cache.GetLatestHash()
-
-	return &walletrpc.BlockID{Height: uint64(latestBlock), Hash: latestHash}, nil
+	bestBlockHash, err := hex.DecodeString(blockChainInfo.BestBlockHash)
+	if err != nil {
+		return nil, err
+	}
+	return &walletrpc.BlockID{Height: uint64(blockChainInfo.Blocks), Hash: []byte(bestBlockHash)}, nil
```

Comment:
This broke the API.

Prior to this, the latest block hash was obtained from the cache, which in turn obtained it from a `CompactBlock`, which in turn obtained its hash from directly hashing a block header:
https://github.com/zcash/lightwalletd/blob/6c400288240615ad4c59e748fd876466553c2e3f/parser/block_header.go#L217-L230

After this change, it decodes the big-endian hex string returned by `getblockchaininfo`, and then fails to byte-reverse it to produce the little-endian byte encoding.

I noticed this because Zaino has reproduced the buggy behaviour in its lightwalletd implementation:
https://github.com/AloeareV/zaino/blob/766d828a3652b34dcd75ad76152fd4648f305bd1/zaino-state/src/fetch.rs#L490-L498